### ISSUE TITLE
Files select markers should be colored according to accent color set (eOS 6 Public Beta).

### ISSUE THREAD

**Issue by lilvadim** [elementary/stylesheet#1017]
These blue circles don't match with accent color that was chosen in settings.
![Screenshot from 2021-05-11 10 12 41](https://user-images.githubusercontent.com/32203382/117752811-a2dddf00-b241-11eb-9f1f-975f93521748.png)
Also, I think they should not look like that on mouse hover, it's not understandable design.

**Comment by jeremypw** [comment]
> Also, I think they should not look like that on mouse hover, it's not understandable design.

Could you expand on this? What should they look like in your opinion?

**Comment by lilvadim** [comment]
> Could you expand on this? What should they look like in your opinion?

I guess it should stay same on mouse hover when file is selected. 
Now it behave like this:

**When file is not selected:** 
_Default_ -- No marker
_On hover_ -- Grayish marker 

**When file is selected:** 
_Default_ -- Blue circle with white sign inside
****_On hover_** -- Blue circle with dark sign inside (why?)**

So I guess it must be like this: 
**When file is selected:** 
_Default_ -- Blue circle with white sign in it 
**_On hover_ -- the same (or just light up a little like files' thumbnails do on hover)**

**Comment by jeremypw** [comment]
OK, so the issue is not related to unselected files?  I'll check the code but this is probably an issue of the stylesheet rather than Files.

**Comment by lilvadim** [comment]
> In any case, the hover issue needs to be raised separately from the accent color issue.

It seems like it's not an issue but looks weird.

**Comment by lilvadim** [comment]
Related: elementary/files#1515

**Comment by danrabbit** [comment]
The selection helpers are icons currently. This could probably be solved by using specially styled CheckButtons instead but I'm not sure how much of a pain it would be to do that @jeremypw ?

**Comment by jeremypw** [comment]
I can look into using Gtk.StyleContext.render_check () instead of Gtk.StyleContext.render_icon () for the helper.

**Comment by jeremypw** [comment]
The effect of rendering a frame on hovered items and a check when selected.  (Just using style classes Gtk.StyleClass.CHECK and Ctk.StyleClass.FRAME and Gtk.StateFlag.CHECKED if selected). 

![RenderCheck](https://user-images.githubusercontent.com/10513844/118043729-aa9aa200-b36d-11eb-8d4f-8f1356d33c6c.png)

**Comment by jeremypw** [comment]
We cannot use CheckButton widgets directly as we are drawing everything ourselves with a CellRenderer. However, we should be able to get the same appearance with judicious modification of the stylecontext. The above is just to show that we do not have to use icons.

**Comment by lilvadim** [comment]
> We cannot use CheckButton widgets directly as we are drawing everything ourselves with a CellRenderer. However, we should be able to get the same appearance with judicious modification of the stylecontext. The above is just to show that we do not have to use icons.

I'm not really into development of eOS, but what about checkboxes that used here: 
![Screenshot from 2021-05-13 20 12 53](https://user-images.githubusercontent.com/32203382/118130561-b34fae80-b427-11eb-8220-9befa38d8aa4.png)

**Comment by jeremypw** [comment]
Those are widgets. The aim would be to emulate their appearance though.

**Comment by danrabbit** [comment]
I think what I'll do is work on using check buttons in switchboard so that the styles are defined and then maybe we can revisit this in Files. I'm hesitant if it's worth doing too much custom drawing code when Gtk4 seems around the corner

**Comment by jeremypw** [comment]
No, I wasn't proposing to add more drawing code - just add different classes to the stylecontext.  Just need the css classes to be available to get the desired appearance (colors, border-radius etc).

**Comment by jeremypw** [comment]
As far as I can see the IconView in Gtk4 still uses a model and renderers - the items are not widgets.  Is there an alternative base widget in Gtk4.  The problem with using widgets to represent files is you would need to create potentially >10^4 of them unless you recycle them (which is complicated).

**Comment by danrabbit** [comment]
@jeremypw I think FlowBox is the replacement for IconView. I believe there was work done to make ListBox and FlowBox more performant

**Comment by jeremypw** [comment]
It would certainly be good to move away from IconView/TreeView and CellRenderers if it there were a scalable widget based replacement!

**Comment by 1280px** [comment]
@danirabbit any progress on this one?

**Comment by danirabbit** [comment]
@1280px please don't create noise in the issue tracker. If there was progress there would be a comment or a linked branch or something. Likely this is blocked by a Gtk 4 port of Files
